### PR TITLE
fix: Allow option values containing quotation marks to be selected

### LINKED ISSUE #29213: cy.select() fails if option value contains a quotation mark
### Current behavior

For an option: 
```js
 <select id="select-maps" name="maps">
  <option value="&quot;test&quot;">test</option>
</select>
```

When you attempt:

```js
cy.get('select[name=maps]').select('"test"')
```

The result is:

```
Timed out retrying after 4000ms: Syntax error, unrecognized expression: option[value=""test""]
```

### Desired behavior

Option is selected

### Test code to reproduce

Incoming pull request with test case

### Cypress Version

13.7.1

### Node version

v18.17.1

### Operating System

macOS 13.6

### Debug Logs

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by davidfurey**
- Closes https://github.com/cypress-io/cypress/issues/29213

### Additional details

I believe this issue was introduced by https://github.com/cypress-io/cypress/pull/25016

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by jennifer-shehane** [comment]
@davidfurey Could you add a changelog entry to this PR? Thanks! https://github.com/cypress-io/cypress/blob/develop/guides/writing-the-cypress-changelog.md

**Comment by davidfurey** [comment]
> @davidfurey Could you add a changelog entry to this PR? Thanks! https://github.com/cypress-io/cypress/blob/develop/guides/writing-the-cypress-changelog.md

Does this belong in `cli/CHANGELOG.md`?

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]
@davidfurey For the changelog, this should be added to the `cli/changelog.md` file, not the file within the main changelog. (This isn't used anymore). 


I verified that this fixes the previously failing test. Thanks for the contribution!

### Before
Screenshot 2024-03-28 at 10 12 23 AM

### After
Screenshot 2024-03-28 at 10 46 07 AM

**Comment by davidfurey** [comment]
Thanks for the speedy review @jennifer-shehane

I've updated the changelog

**Review by jennifer-shehane — APPROVED** [review]

**Review by cacieprins — APPROVED** [review]

File: packages/driver/src/cy/commands/actions/select.ts [hunk 1]
```diff
@@ -128,7 +128,7 @@ export default (Commands, Cypress, cy) => {
             values.push(value)
 
             // https://github.com/cypress-io/cypress/issues/24739
-            if (options.$el.find(`option[value="${value}"]`).length > 1) {
+            if (options.$el.find(`option[value="${value.replace(/"/g, '\\\"')}"]`).length > 1) {
```
Comment (samelawrence) [inline packages/driver/src/cy/commands/actions/select.ts:131]: @davidfurey Only if you have the spare cycles - but I'm curious as a RegEx noob what this logic actually does and how it resolves the bug. Would appreciate an explanation (but again, only if you have time). Thank you!

**Comment by jennifer-shehane** [comment]
Going to merge with these chrome beta tests failing, this is a known issue we're investigating here: https://github.com/cypress-io/cypress/issues/29199

**Pull request merged**